conda_lock/src_parser/aggregation.py

```diff
@@ -51,3 +54,34 @@ def aggregate_lock_specs(
             lock_spec.allow_pypi_requests for lock_spec in lock_specs
         ),
     )
+
+
+def unify_package_sources(collections: List[List[Channel]]) -> List[Channel]:
+    """Unify the package sources from multiple lock specs.
+
+    To be able to merge the lock specs, the package sources must be compatible between
+    them. This means that between any two lock specs, the package sources must be
+    identical or one must be an extension of the other.
+
+    This allows us to use a superset of all of the package source lists in the
+    aggregated lock spec.
+
+    The following is allowed:
+
+    > unify_package_sources([[channel_two, channel_one], [channel_one]])
+    [channel_two, channel_one]
+
+    Whilst the following will fail:
+
+    > unify_package_sources([[channel_two, channel_one], [channel_three, channel_one]])
+
+    In the failing example, it is not possible to predictably decide which channel
+    to search first, `channel_two` or `channel_three`, so we error in this case.
+    """
+    if not collections:
+        return []
+    result = max(collections, key=len)
+    for collection in collections:
+        if collection != result[-len(collection) :]:
```

Comment:
This still suffers from the edge case I mentioned earlier:

```
unify_package_sources([[], [1,2,3]])

ValueError: [] is not an ordered subset of [1, 2, 3]
```

because `result[-0 :]` is the same as `result[0:]` which is the whole list.